qiskit_machine_learning/neural_networks/sampler_qnn.py

```diff
@@ -317,26 +331,53 @@ def _postprocess_gradient(
 
         return input_grad, weights_grad
 
+    def _forward(
+        self,
+        input_data: np.ndarray | None,
+        weights: np.ndarray | None,
+    ) -> np.ndarray | SparseArray | None:
+        """
+        Forward pass of the network.
+        """
+        parameter_values, num_samples = self._preprocess(input_data, weights)
+
+        if num_samples is not None and np.prod(parameter_values.shape) > 0:
+            # sampler allows batching
+            job = self.sampler.run([self._circuit] * num_samples, parameter_values)
+            results = job.result()
+        else:
+            results = None
+
+        result = self._postprocess(num_samples, results)
+
+        return result
```

Comment:
I guess I find the above "awkward" somehow. To me it would be more transparent what it was doing overall were it more like
```
            result = self._postprocess(num_samples, job.result())
        else:
            result = None

        return result
```
Explicitly passing None into postprocess as results and having it internally shortcircuit its processing and return None just seems a bit convoluted and took more to follow. Maybe there is some rationale that its easier to have it check in one place in postprocess but for me it made things harder to follow here.